File: 프로그래머스/unrated/135807. 숫자 카드 나누기/숫자 카드 나누기.py
from math import gcd

def check(arrA, arrB):
    c1 = arrA[0]
    for n in arrA:
        if (c1:=gcd(c1, n)) == 1:
            return 0
    for c in range(c1,1,-1):
        if c1%c==0 and all(n%c for n in arrB):
            return c
    return 0

def solution(arrayA, arrayB):
    return max(check(arrayB, arrayA), check(arrayA, arrayB))
            
        
    

File: 프로그래머스/unrated/135807. 숫자 카드 나누기/숫자 카드 나누기.js
const _gcd = (a, b) => (b ? _gcd(b, a % b) : a)

const check = (arrA, arrB) => {
  g = arrA.reduce((acc, cur) => _gcd(acc, cur))
  for (let n = g; n > 0; n--) {
    if (g % n == 0 && arrB.every(v => v % n)) return n
  }
  return 0
}

const solution = (arrayA, arrayB) =>
  Math.max(check(arrayA, arrayB), check(arrayB, arrayA))